packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputMultipleValue.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Autocomplete, { AutocompleteProps } from '@mui/material/Autocomplete';
+import Chip from '@mui/material/Chip';
+
+import TextField from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridFilterItem } from '../../../models/gridFilterItem';
+
+export type GridFilterInputMultipleValueProps = {
+  item: GridFilterItem;
+  applyValue: (value: GridFilterItem) => void;
+  // Is any because if typed as GridApiRef a dep cycle occurs. Same happens if ApiContext is used.
+  apiRef: any;
+  focusElementRef?: React.Ref<any>;
+  type?: 'text' | 'number';
+} & Omit<AutocompleteProps<any[], true, false, true>, 'options' | 'renderInput'>;
+
+function GridFilterInputMultipleValue(props: GridFilterInputMultipleValueProps) {
+  const { item, applyValue, type, apiRef, focusElementRef, ...other } = props;
+  const [filterValueState, setFilterValueState] = React.useState(item.value || []);
+  const id = useId();
+
+  React.useEffect(() => {
+    const itemValue = item.value ?? [];
+    setFilterValueState(itemValue.map(String));
+  }, [item.value]);
+
+  const onFilterChange = React.useCallback(
```

Comment:
To keep with the React convention.

```suggestion
  const handleChange = React.useCallback(
```